SofaKernel/modules/SofaHelper/src/sofa/helper/system/PluginManager.cpp

```diff
@@ -321,12 +321,19 @@ std::string PluginManager::findPlugin(const std::string& pluginName, const std::
     const std::string libName = DynamicLibrary::prefix + name + "." + DynamicLibrary::extension;
 
     // First try: case sensitive
-    for (std::vector<std::string>::iterator i = searchPaths.begin(); i!=searchPaths.end(); i++)
-    {
-        const std::string path = *i + "/" + libName;
-        if (FileSystem::isFile(path))
-            return path;
+    for (const auto & prefix : searchPaths) {
+        const std::array<std::string, 3> paths = {
+                prefix + "/" + libName,
+                prefix + "/" + pluginName + "/" + libName,
+                prefix + "/" + pluginName + "/lib/" + libName
```

Comment:
And what about the other 69.5% of our users on Windows? :smile:

Reply:
It should be fine, no? They will get caught into the first `prefix + "/" + libName` (which is exactly the same path as before). The other two paths should be quickly discarded as non-existent.

Reply:
Nope, plugins are located in `$SOFA_ROOT/plugins/PLUGIN_NAME/bin/PLUGIN_NAME.dll` so, while non-breaking, this block will not help Windows users.

Reply:
Haaa ok, I though there were no such things as relocatable plugins in Windows. I'll add this then.